src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -101,6 +115,14 @@ space use `euclidean_space 𝕜 (fin n)`. -/
 def euclidean_space (𝕜 : Type*) [is_R_or_C 𝕜]
   (n : Type*) [fintype n] : Type* := pi_Lp 2 (λ (i : n), 𝕜)
 
+/-- The (forgetful) equivalence between `euclidean_space 𝕜 ι` and maps, `ι → 𝕜`:
+backwards direction. -/
+def to_euclidean_space [fintype ι] : (ι → 𝕜) ≃ euclidean_space 𝕜 ι := equiv.refl _
+
+/-- The (forgetful) equivalence between `euclidean_space 𝕜 ι` and maps, `ι → 𝕜`:
+forwards direction. -/
+def of_euclidean_space [fintype ι]: euclidean_space 𝕜 ι ≃ (ι → 𝕜) := equiv.refl _
+
```

Comment:
These are going to make things annoying unless you add a handful of simp lemmas about them too. I recommend:

* Adding `simp` lemmas stating:
  * `to_euclidean_space (of_euclidean_space e) = e`
  * `of_euclidean_space (to_euclidean_space x) = x`
  * `to_euclidean_space x i = x i`
  * `of_euclidean_space e i = e i`
* Either:
  * Making these `linear_equivs` over `𝕜` / `add_equiv`s
  * Adding `_add`, `_neg`, `_sub`, `_zero`, `_smul` lemmas for both

Reply:
I think we just shouldn't have these.  See https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/orthonormal.20basis.20structure/near/272880245

Reply:
I guess either way, they probably don't need to be part of _this_ PR, although my hunch is we will want them eventually